src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -85,15 +96,20 @@ module Make (Inputs : Inputs_intf) :
     Transition_handler.Validator.run ~frontier
       ~transition_reader:network_transition_reader ~valid_transition_writer
       ~logger ;
+    List.iter collected_transitions
```

Comment:
If this list is longer than the `capacity`, we're going to drop a lot of messages here. That doesn't seem like the right behavior -- what should we do instead?

Reply:
I think we have already handled this. When we define the capacity (primary_transition_pipe_capacity), it is defined to the size of all the collected_transitions + the capacity of previous pipe. So we wouldn't drop a lot of messages I believe.

Reply:
I see, I should have looked closer :)